### ISSUE TITLE
Expected override method from class will not warning `invalid-name`

### ISSUE THREAD

**Issue by daoiqi** [PyCQA/pylint#4183]
### Steps to reproduce

```python
#!/usr/bin/env python
# --coding:utf-8--

# pylint: disable=missing-module-docstring,missing-class-docstring,missing-function-docstring

from http.server import BaseHTTPRequestHandler, HTTPServer


class TestHTTPServerRequestHandler(BaseHTTPRequestHandler):
    def do_GET(self):
        self.send_error(500, "not found")


def run():
    port = 8000
    print("starting server, port", port)
    # Server settings
    server_address = ("", port)
    httpd = HTTPServer(server_address, TestHTTPServerRequestHandler)
    print("running server...")
    httpd.serve_forever()


if __name__ == "__main__":
    run()
```

### Current behavior

Result of `pylint a.py`:
```
pylint test2.py
************* Module test2
test2.py:10:4: C0103: Method name "do_GET" doesn't conform to snake_case naming style (invalid-name)

------------------------------------------------------------------
Your code has been rated at 9.23/10 (previous run: 6.15/10, +3.08)

```

### Expected behavior

Expected override method from class will not warning `invalid-name`

such as  `do_GET` `do_POST` `runTest` 

### pylint --version output

Result of `pylint --version` output:
```
pylint --version output
pylint 2.7.2
astroid 2.5.1
Python 3.8.5 (default, Sep  4 2020, 02:22:02)
[Clang 10.0.0 ]
```

**Comment by dbaty** [comment]
This is similar to #4060.

**Comment by Pierre-Sassoulas** [comment]
Closing as duplicate of #4563. A temporary fix exists in https://github.com/PyCQA/pylint/issues/4060#issuecomment-774438060 (you can modify the regex to include your function name)
